### ISSUE TITLE
Regular expression trouble

### ISSUE THREAD

**Issue by sergeizhukov** [codemix/yii2-localeurls#153]
Hello, can you help me?
I have this rule:

```

        'urlManager' => [
            'class' => codemix\localeurls\UrlManager::class,
            'languages' => [
                'de' => 'de',
                'fr' => 'fr',
            ],

            'showScriptName' => false,
            'enablePrettyUrl' => true,
            'rules' => [
                'artikel/<id:\d+>-{0,1}<slug:[\w-]+>{0,1}' => 'article/view',
                '<controller:[\w-]+>/<id:\d+>'=>'<controller>/view',
                '<controller:[\w-]+>/<action:[\w-]+>/<id:\d+>'=>'<controller>/<action>',
                '<controller:[\w-]+>/<action:[\w-]+>'=>'<controller>/<action>',
            ],
            'ignoreLanguageUrlPatterns' => [
                '#^site/change-lang#' => '#^site/change-lang#',
            ],
        ],
        
```

When i go to default url http://site.com/artikel/6-slug
It works.
But if i will go with http://site.com/fr/artikel/6-slug
I will see this in browser and page not found:
`http://site.com/fr/artikel/6-%7B0,1%slug%7B0,1%7D`

The regular expression was fallen to url
Any ideas?

**Comment by mikehaertl** [comment]
Can you explain a bit more? What do you mean "if i will go with <url>"? Do you copy this URL into the browser? This URL already has a language code, so this extension will not create a redirect. If you see a redirect to the broken URL this must be generated somewhere else.

What I would do:
* Enable `YII_DEBUG` and check the logs. Our extension creates some useful messages.
* Try `curl -I http://site.com/fr/artikel/6-slug` and inspect if there's a `Location:` header sent and what it contains

**Comment by sergeizhukov** [comment]
Hm now the problem disappear. Magic. oO Just Url in browser showed me part of regular expression.
But I have some other question:
If i'm right now on `de` version, and if going to http://site.com/fr/artikel/6-slug. Should the `_language` cookie set the `fr` language?
Because I do
`var_dump(\Yii::$app->getRequest()->getCookies()->get('_language')); exit;`
And see
`yii\webCookie (object) [Object ID #89][7 properties]
name: (string) "_language"
value: (string) "de"
domain: (string) ""
expire: (null) NULL
path: (string) "/"
secure: (boolean) false 
httpOnly: (boolean) true `

**Comment by mikehaertl** [comment]
Cookie should always be updated, yes. All this is standard behavior and I'm pretty sure it works as we would have many more issue reports. It's also covered by our test cases.

I'm closing this for now unless you can provide a clear example for what is broken and how to reproduce it.

**Comment by mikehaertl** [comment]
One more note: Again, please check the Trace-Logs. They will tell you, when a cookie gets updated or when a redirect is sent.

**Comment by sergeizhukov** [comment]
I just types in browser `site.com/fr` It works only for current page.
From server no response cookies. 
http://site.com/artikel/6-slug?language=fr
This url still render german for me. 
Maybe wrong configuration.
But I resolved it:
I added `'enableLanguagePersistence' => false,` in the config.
Before request I just restore the user language:

```
$lang = \Yii::$app->session->get('lang');

        if ($lang) {
            \Yii::$app->language = $lang;
        }
```
Now it works on both languages is correct.
